bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/i18n/ThingTypeI18nLocalizationService.java

```diff
@@ -69,126 +64,21 @@ protected void unsetTranslationProvider(TranslationProvider i18nProvider) {
-                ChannelType channelType = channelTypeRegistry.getChannelType(channelTypeUID, locale);
-                if (channelType != null) {
-                    if (channelLabel == null) {
-                        channelLabel = thingTypeI18nUtil.getChannelLabel(bundle, channelTypeUID, channelType.getLabel(),
-                                locale);
-                    }
-                    if (channelDescription == null) {
-                        channelDescription = thingTypeI18nUtil.getChannelDescription(bundle, channelTypeUID,
-                                channelType.getDescription(), locale);
-                    }
-                }
-            }
-            localizedChannelDefinitions
-                    .add(new ChannelDefinition(channelDefinition.getId(), channelDefinition.getChannelTypeUID(),
-                            channelDefinition.getProperties(), channelLabel, channelDescription));
-        }
-        return localizedChannelDefinitions;
+    @Reference
+    protected void setChannelI18nUtil(ChannelI18nUtil channelI18nUtil) {
+        this.channelI18nUtil = channelI18nUtil;
     }
 
-    public ChannelGroupType createLocalizedChannelGroupType(Bundle bundle, ChannelGroupType channelGroupType,
-            @Nullable Locale locale) {
-        ChannelGroupTypeUID channelGroupTypeUID = channelGroupType.getUID();
-        String defaultLabel = channelGroupType.getLabel();
-        String label = thingTypeI18nUtil.getChannelGroupLabel(bundle, channelGroupTypeUID, defaultLabel, locale);
-        String description = thingTypeI18nUtil.getChannelGroupDescription(bundle, channelGroupTypeUID,
-                channelGroupType.getDescription(), locale);
-
-        List<ChannelDefinition> localizedChannelDefinitions = createLocalizedChannelDefinitions(bundle,
-                channelGroupType.getChannelDefinitions(),
-                channelDefinition -> thingTypeI18nUtil.getChannelLabel(bundle, channelGroupTypeUID, channelDefinition,
-                        channelDefinition.getLabel(), locale),
-                channelDefinition -> thingTypeI18nUtil.getChannelDescription(bundle, channelGroupTypeUID,
-                        channelDefinition, channelDefinition.getDescription(), locale),
-                locale);
-
-        ChannelGroupTypeBuilder builder = ChannelGroupTypeBuilder
-                .instance(channelGroupTypeUID, label == null ? defaultLabel : label)
-                .isAdvanced(channelGroupType.isAdvanced()).withCategory(channelGroupType.getCategory())
-                .withChannelDefinitions(localizedChannelDefinitions);
-        if (description != null) {
-            builder.withDescription(description);
-        }
-        return builder.build();
+    protected void unsetChannelI18nUtil(ChannelI18nUtil channelI18nUtil) {
+        this.channelI18nUtil = null;
     }
 
     private List<ChannelGroupDefinition> createLocalizedChannelGroupDefinitions(final Bundle bundle,
```

Comment:
I only did the ChannelI18nUtil because the `createLocalizedChannelDefinitions` method is used in both `ThingTypeI18nLoc...` and `ChannelGroupTypeI18nLoc...`.
I would like to keep further refactorings out of this PR and consolidate, move, clean up in a separate PR.